xbmc/Util.cpp

```diff
@@ -1467,8 +1467,10 @@ bool CUtil::MakeShortenPath(std::string StrInput, std::string& StrOutput, size_t
     nGreaterDelim = nPos;
     if ( nPos != std::string::npos )
       nPos = StrInput.find_last_of( cDelim, nPos - 1 );
-    if ( nPos == std::string::npos ) break;
-    if ( nGreaterDelim > nPos ) StrInput.replace( nPos + 1, nGreaterDelim - nPos - 1, ".." );
+    if ( nPos == 0 )
```

Comment:
On libreelec the following steps (you need to have "Expert" settings enabled):

Go to "Settings" -> "Player" -> "Discs" -> "Saved music folder" ->
Select a long folder, where you see the "/../../" -> Press OK.

Not all lengths bring the problem, sometimes on much bigger it works.

But e.g. this bring the fault:
"/storage/videos/Star_Trek_Das_nächste_Jahrhundert"